Add tests for fractionation module

daklauss opened the pull request:
Added the Cases i used for the fractionating topic as test to have a small suit that checks wheter the standard ```fractionoptimizer``` and it's default parameter are suitable. Hopefully this shows possible instabilities with the cobyla optimizer when changing parameters.

I guess the cobyla fixture is a litle bit messy. 

Also changed the parameters for tol and rhoberg another time.

schmoelder committed:
Add fractionation optimizer test suite

Co-authored-by: daklauss <d.klauss@fz-juelich.de>

Commit:
Add fractionating test

Commit:
Play with COBYLA parameters

schmoelder commented on tests/test_fractionating.py:
```diff
@@ -0,0 +1,296 @@
+    """Class to create a faked SolutionIO object with a Gauss function."""
+
+    def __init__(self, solution_struct, comp_system):
+        """solution struct has to be a dictionary with
```
Improve docstring

schmoelder commented on tests/test_fractionating.py:
Consider a different name. "Fractionating" is not a really a thing. Consider what exactly you are testing here. Is it the FractionationOptimizer? Is it performance?

schmoelder commented on CADETProcess/fractionation/fractionationOptimizer.py:
Does this make the optimization more robust or is this just "to play around"?

schmoelder committed:
Extract synthetic chromatogram helpers to shared test modules

The SolutionIO builders are needed across fractionation and comparison tests; keeping them in a single test file made reuse awkward.

schmoelder committed:
Use chromatogram factory in comparison tests

The factory needed t_end, n_time, and heights parameters to cover the 0–100 time range and mixed-height signals in the comparison suite.
